src/query/parser/promql/parse.go

```diff
@@ -291,17 +291,27 @@ func (p *parseState) walk(node pql.Node) error {
 			argType := argTypes[i]
 			expr := expressions[i]
 			if argType == pql.ValueTypeScalar {
-				val, err := resolveScalarArgument(expr)
+				val, err := resolveScalarArgument(unwrapParenExpr(expr))
 				if err != nil {
 					return err
 				}
 
 				argValues = append(argValues, val)
 			} else if argType == pql.ValueTypeString {
-				stringValues = append(stringValues, expr.(*pql.StringLiteral).Val)
+				val, err := resolveStringArgument(unwrapParenExpr(expr))
+				if err != nil {
+					return err
+				}
+
+				stringValues = append(stringValues, val)
 			} else {
-				if e, ok := expr.(*pql.MatrixSelector); ok {
-					argValues = append(argValues, e.Range)
+				if argType == pql.ValueTypeMatrix {
+					val, err := resolveMatrixRangeArgument(unwrapParenExpr(expr))
```

Comment:
I'm afraid this changes semantics a little. Previously, if `.(*pql.MatrixSelector)` type assertion failed, nothing would get appended to `argValues`. With this change, `time.Duration(0)` will get appended to it.
Do we understand if that would be OK?

Reply:
True, missed that.

It seems that the parentheses unwrapping was already added in a previous commit, a few lines above:
https://github.com/m3db/m3/blob/1c7bb7ed9806e8a311fed69d1bf8ad93b7085ace/src/query/parser/promql/parse.go#L255-L257

I've restored the code https://github.com/m3db/m3/pull/2809/commits/ebacef9444d64b65c861f6442b0e9fa4e017a79d